Remove threading assertion in copyLocalizedString

achristensen07 opened the pull request:
#### 7090e844eec54fcae5a538606c06165818a0f990
<pre>
Remove threading assertion in copyLocalizedString
<a href="https://bugs.webkit.org/show_bug.cgi?id=286385">https://bugs.webkit.org/show_bug.cgi?id=286385</a>
<a href="https://rdar.apple.com/143431906">rdar://143431906</a>

Reviewed by David Kilzer.

The function seems thread safe, and if you run accessibility layout tests with --accessibility-isolated-tree
you get several assertions in that function, such as stacks like this:

WTFCrashWithInfo(int, char const*, char const*, int) + 100
WebCore::copyLocalizedString(__CFString const*) + 148 (LocalizedStrings.cpp:105)
WebCore::localizedString(__CFString const*) + 44 (LocalizedStrings.cpp:127)
-[WebAccessibilityObjectWrapperBase accessibilityCustomContent] + 332 (WebAccessibilityObjectWrapperBase.mm:414)
WTR::AccessibilityUIElement::customContent() const::$_0::operator()() const + 96 (AccessibilityUIElementMac.mm:504)
WTF::Detail::CallableWrapper&lt;WTR::AccessibilityUIElement::customContent() const::$_0, void&gt;::call() + 32 (Function.h:53)
WTF::Function&lt;void ()&gt;::operator()() const + 184 (Function.h:82)
WTR::AccessibilityController::executeOnAXThreadAndWait(WTF::Function&lt;void ()&gt;&amp;&amp;)::$_0::operator()() const + 36 (AccessibilityController.cpp:117)
WTF::Detail::CallableWrapper&lt;WTR::AccessibilityController::executeOnAXThreadAndWait(WTF::Function&lt;void ()&gt;&amp;&amp;)::$_0, void&gt;::call() + 32 (Function.h:53)
WTF::Function&lt;void ()&gt;::operator()() const + 184 (Function.h:82)
WTR::AXThread::dispatchFunctionsFromAXThread() + 260 (AccessibilityController.cpp:256)
WTR::AXThread::threadRunLoopSourceCallback() + 48 (AccessibilityControllerMac.mm:207)
WTR::AXThread::threadRunLoopSourceCallback(void*) + 28 (AccessibilityControllerMac.mm:201)

* Source/WebCore/platform/LocalizedStrings.cpp:
(WebCore::copyLocalizedString):

Canonical link: <a href="https://commits.webkit.org/289272@main">https://commits.webkit.org/289272@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f511c0485338d268e67f4799f9f9336f8ddc6b41

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/86237 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/5805 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/40575 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/91187 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/37076 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/88288 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/6053 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/13839 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/91187 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/37076 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/89242 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/4610 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/78138 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/91187 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/4425 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/32448 "Passed tests") | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/36173 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/74991 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/33314 "Passed tests") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/92987 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/13458 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/9800 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/92987 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/13671 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/74009 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/92987 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/19024 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/17419 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/6254 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/13414 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/13480 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/18794 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/13248 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/16681 "Built successfully") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/124/builds/15034 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

ddkilzer reviewed (approved):
r=me

aproskuryakov commented on Source/WebCore/platform/LocalizedStrings.cpp:
```diff
@@ -99,12 +99,6 @@ static CFBundleRef webCoreBundle()
-
     static CFStringRef notFound = CFSTR("localized string not found");
 
     auto result = adoptCF(CFBundleCopyLocalizedString(webCoreBundle(), key, notFound, nullptr));
```
webCoreBundle() uses NeverDestroyed, is that actually thread safe?

achristensen07 replied:
Good point.  I think the answer is technically no, but I think if multiple threads tried to initialize it at the same time, any corruption would probably not cause any problems in practice.  May as well make it really thread safe, though.

achristensen07 committed:
Remove threading assertion in copyLocalizedString
https://bugs.webkit.org/show_bug.cgi?id=286385
rdar://143431906

Reviewed by David Kilzer.

The function seems thread safe, and if you run accessibility layout tests with --accessibility-isolated-tree
you get several assertions in that function, such as stacks like this:

WTFCrashWithInfo(int, char const*, char const*, int) + 100
WebCore::copyLocalizedString(__CFString const*) + 148 (LocalizedStrings.cpp:105)
WebCore::localizedString(__CFString const*) + 44 (LocalizedStrings.cpp:127)
-[WebAccessibilityObjectWrapperBase accessibilityCustomContent] + 332 (WebAccessibilityObjectWrapperBase.mm:414)
WTR::AccessibilityUIElement::customContent() const::$_0::operator()() const + 96 (AccessibilityUIElementMac.mm:504)
WTF::Detail::CallableWrapper<WTR::AccessibilityUIElement::customContent() const::$_0, void>::call() + 32 (Function.h:53)
WTF::Function<void ()>::operator()() const + 184 (Function.h:82)
WTR::AccessibilityController::executeOnAXThreadAndWait(WTF::Function<void ()>&&)::$_0::operator()() const + 36 (AccessibilityController.cpp:117)
WTF::Detail::CallableWrapper<WTR::AccessibilityController::executeOnAXThreadAndWait(WTF::Function<void ()>&&)::$_0, void>::call() + 32 (Function.h:53)
WTF::Function<void ()>::operator()() const + 184 (Function.h:82)
WTR::AXThread::dispatchFunctionsFromAXThread() + 260 (AccessibilityController.cpp:256)
WTR::AXThread::threadRunLoopSourceCallback() + 48 (AccessibilityControllerMac.mm:207)
WTR::AXThread::threadRunLoopSourceCallback(void*) + 28 (AccessibilityControllerMac.mm:201)

* Source/WebCore/platform/LocalizedStrings.cpp:
(WebCore::copyLocalizedString):

Canonical link: https://commits.webkit.org/289272@main
